p2p/net/swarm/swarm_dial.go

```diff
@@ -372,6 +380,9 @@ func (s *Swarm) resolveAddrs(ctx context.Context, pi peer.AddrInfo) ([]ma.Multia
 			}
 			var added bool
 			for _, a := range resolvedAddrs {
+				if len(toResolve) >= maximumToResolveAddresses {
+					break
+				}
 				if !addr.Equal(a) {
 					toResolve = append(toResolve, a)
 					added = true
```

Comment:
Do we need to maintain this variable `added`? When this block ends, we want to continue with the outer loop, as we have resolved the variable with the transport, right?

The current code says, if the resolver didn't modify the address, we should try to resolve it using our madns resolver. Do we want that? If so, can you add a comment for it?